File: patch_doer_profile.py
import re

with open('src/components/DoerProfileModal.tsx', 'r') as f:
    content = f.read()

# Add viewingImage state
state_code = """  const [viewingImage, setViewingImage] = useState<string | null>(null);"""

if 'setViewingImage' not in content:
    content = content.replace("const [showEduTooltip, setShowEduTooltip] = useState(false);", f"const [showEduTooltip, setShowEduTooltip] = useState(false);\n{state_code}")

# Update Cover image to be clickable
cover_old = """        {/* PROFILE COVER BANNER */}
        <div className="h-40 md:h-48 w-full relative bg-slate-200 overflow-hidden">
          {profile.coverImageUrl ? (
            <img src={profile.coverImageUrl} alt="Cover" className="absolute inset-0 w-full h-full object-cover" />
          ) : ("""

cover_new = """        {/* PROFILE COVER BANNER */}
        <div className="h-40 md:h-48 w-full relative bg-slate-200 overflow-hidden">
          {profile.coverImageUrl ? (
            <img onClick={() => setViewingImage(profile.coverImageUrl)} src={profile.coverImageUrl} alt="Cover" className="absolute inset-0 w-full h-full object-cover cursor-pointer" />
          ) : ("""
content = content.replace(cover_old, cover_new)

# Update Avatar to be clickable
avatar_src_logic = "src={profile.profileImageUrl || profile.avatarUrl || (profile.id === 'doer-1' ? 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?w=150&fit=crop&q=80' : profile.id === 'doer-2' ? 'https://images.unsplash.com/photo-1494790108377-be9c29b29330?w=150&fit=crop&q=80' : profile.id === 'doer-4' ? 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?w=150&fit=crop&q=80' : 'https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?w=150&fit=crop&q=80')}"
avatar_alt = "alt={isOwnProfile ? `${currentUser.firstName} ${currentUser.lastName}` : profile.title}"

avatar_old = f"""            <div className="w-20 h-20 md:w-24 md:h-24 rounded-full overflow-hidden border-4 border-white shadow-lg relative bg-white flex-shrink-0">
              <img
                {avatar_src_logic}
                {avatar_alt}
                className="w-full h-full object-cover"
              />
            </div>"""

avatar_new = f"""            <div className="w-20 h-20 md:w-24 md:h-24 rounded-full overflow-hidden border-4 border-white shadow-lg relative bg-white flex-shrink-0">
              <img
                onClick={{(e) => setViewingImage(e.currentTarget.src)}}
                {avatar_src_logic}
                {avatar_alt}
                className="w-full h-full object-cover cursor-pointer"
              />
            </div>"""

content = content.replace(avatar_old, avatar_new)

# Add Lightbox at the end
render_additions = """      <AnimatePresence>
        {viewingImage && (
          <div className="fixed inset-0 z-[120] bg-black/95 flex items-center justify-center p-4">
            <button onClick={() => setViewingImage(null)} className="absolute top-4 right-4 text-white p-2">
              <X className="w-8 h-8" />
            </button>
            <img src={viewingImage} alt="Fullscreen View" className="max-w-full max-h-full object-contain" />
          </div>
        )}
      </AnimatePresence>
    </motion.div>
  );
}"""

content = re.sub(r'    </motion\.div>\n  \);\n}', render_additions, content)

with open('src/components/DoerProfileModal.tsx', 'w') as f:
    f.write(content)

File: patch_doerprofile.js
const fs = require('fs');
let code = fs.readFileSync('src/components/DoerProfileModal.tsx', 'utf8');

code = code.replace(
  "import ConfirmationModal from './ConfirmationModal';",
  "import ConfirmationModal from './ConfirmationModal';\nimport ProfileQRCodeModal from './ProfileQRCodeModal';"
);

code = code.replace(
  "const [activeTab, setActiveTab] = useState('portfolio');",
  "const [activeTab, setActiveTab] = useState('portfolio');\n  const [isQRModalOpen, setIsQRModalOpen] = useState(false);"
);

code = code.replace(
  `        {/* Top Header Controls */}
        <div className="absolute top-4 right-4 z-20 flex gap-2">
          <button
            onClick={() => {
              triggerSound('click');
              handleShare(profile.displayName, \`Check out \${profile.displayName}'s profile on our app!\`);
            }}
            className="p-2.5 bg-slate-900/40 hover:bg-slate-900/60 text-white rounded-full transition-all shadow-md backdrop-blur-xs flex items-center justify-center group cursor-pointer"
            title="Share Profile"
          >`,
  `        {/* Top Header Controls */}
        <div className="absolute top-4 right-4 z-20 flex gap-2">
          <button
            onClick={() => {
              triggerSound('click');
              setIsQRModalOpen(true);
            }}
            className="p-2.5 bg-slate-900/40 hover:bg-slate-900/60 text-white rounded-full transition-all shadow-md backdrop-blur-xs flex items-center justify-center group cursor-pointer"
            title="Show QR Code"
          >
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2.5" strokeLinecap="round" strokeLinejoin="round" className="w-4 h-4 text-white"><rect width="5" height="5" x="3" y="3" rx="1"/><rect width="5" height="5" x="16" y="3" rx="1"/><rect width="5" height="5" x="3" y="16" rx="1"/><path d="M21 16h-3a2 2 0 0 0-2 2v3"/><path d="M21 21v.01"/><path d="M12 7v3a2 2 0 0 1-2 2H7"/><path d="M3 12h.01"/><path d="M12 3h.01"/><path d="M12 16v.01"/><path d="M16 12h1"/><path d="M21 12v.01"/><path d="M12 21v-1"/></svg>
          </button>
          <button
            onClick={() => {
              triggerSound('click');
              handleShare(profile.displayName, \`Check out \${profile.displayName}'s profile on our app!\`);
            }}
            className="p-2.5 bg-slate-900/40 hover:bg-slate-900/60 text-white rounded-full transition-all shadow-md backdrop-blur-xs flex items-center justify-center group cursor-pointer"
            title="Share Profile"
          >`
);

code = code.replace(
  "        {/* Full View Content Layer */}",
  `        {/* Full View Content Layer */}
        
        <ProfileQRCodeModal 
          isOpen={isQRModalOpen} 
          onClose={() => setIsQRModalOpen(false)} 
          profileId={profile.id} 
          profileName={profile.displayName} 
        />`
);

fs.writeFileSync('src/components/DoerProfileModal.tsx', code);
